cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc

```diff
@@ -0,0 +1,359 @@
+// aliases: ['/docs/aws/waf/cloud-front.md']
+// tags: ["AWS", "ROSA"]
+// authors:
+//  - 'Connor Wooley'
+// ---
+
+AWS WAF is a web application firewall that lets you monitor the HTTP and HTTPS requests that are forwarded to your protected web application resources.
+
+You can use an Amazon CloudFront to add a Web Application Firewall (WAF) to your {product-title} (ROSA) workloads. Using an external solution protects ROSA resources from experiencing denial of service due to handling the WAF.
+
+[id="prerequisites_{context}"]
+== Prerequisites
+
+* xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-creating-a-cluster-quickly[A ROSA Classic cluster].
+* You have access to the OpenShift CLI (`oc`).
+* You have access to the AWS CLI (`aws`).
+
+[id="environment-setup_{context}"]
+=== Environment setup
+
+* Prepare the environment variables:
++
+[source,terminal]
+----
+$ export AWS_PAGER=""
+$ export CLUSTER_NAME=$(oc get infrastructure cluster -o=jsonpath="{.status.infrastructureName}"  | sed 's/-[a-z0-9]\{5\}$//')
+$ export REGION=$(oc get infrastructure cluster -o=jsonpath="{.status.platformStatus.aws.region}")
+$ export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+$ export SCRATCH="/tmp/${CLUSTER_NAME}/cloudfront-waf"
+$ mkdir -p ${SCRATCH}
+$ echo "Cluster: ${CLUSTER_NAME}, Region: ${REGION}, AWS Account ID: ${AWS_ACCOUNT_ID}"
+----
+
+[id="custom_domain_setup{context}"]
+== Custom domain setup
+
+It is necessary to configure a secondary ingress controller to segment your external WAF-protected traffic from your standard (and default) cluster ingress controller. In ROSA, we do this using the Custom Domain Operator.
+
+.Prerequisites
+
+* A unique domain, such as `*.apps.<company_name>.io`
+* A custom SAN or wildcard certificate, such as `CN=*.apps.<company_name>.io`
+
+.Procedure 
+
+. Create a new project
++
+[source,terminal]
+----
+$ oc new-project waf-demo
+----
+
+. Create a new TLS secret from a private key and a public certificate, where `fullchain.pem` is your full wildcard certificate chain (including any intermediaries) and `privkey.pem` is your wildcard certificate's private key.
++
+.Example
+[source,terminal]
+----
+$ oc -n waf-demo create secret tls waf-tls --cert=fullchain.pem --key=privkey.pem
+----
+
+. Create a new `CustomDomain` custom resource (CR):
++
+.Example `waf-custom-domain.yaml`
+[source,yaml]
+----
+apiVersion: managed.openshift.io/v1alpha1
+kind: CustomDomain
+metadata:
+  name: cloudfront-waf
+spec:
+  domain: apps.<company_name>.io <1>
+  scope: External
+  loadBalancerType: NLB
+  certificate:
+    name: waf-tls
+    namespace: waf-demo
+  routeSelector: <2>
+    matchLabels:
+     route: waf
+----
+<1> The custom domain.
+<2> Filters the set of routes serviced by the CustomDomain ingress. In this tutorial, we will use the `waf` route selector, but if no value was to be provided, no filtering would occur.
```

Comment:
Suggestion: 
<2> Filters the set of routes serviced by the CustomDomain Ingress. This tutorial uses the `waf` route selector, but if no value was to be provided, no filtering would occur.